test/integration/services/policy_engine/test_namespaces.py

```diff
@@ -0,0 +1,182 @@
+        DistroNamespace(name='ol', version='7.3', like_distro='ol'),
+        DistroNamespace(name='rhel', version='6', like_distro='rhel'),
+        DistroNamespace(name='rhel', version='7', like_distro='rhel'),
+        DistroNamespace(name='rhel', version='7.1', like_distro='rhel'),
+        DistroNamespace(name='rhel', version='8', like_distro='rhel'),
+        DistroNamespace(name='rhel', version='8.1', like_distro='rhel'),
+        DistroNamespace(name='debian', version='8', like_distro='debian'),
+        DistroNamespace(name='debian', version='9', like_distro='debian'),
+        DistroNamespace(name='debian', version='10', like_distro='debian'),
+        DistroNamespace(name='debian', version='11', like_distro='debian'),
+        DistroNamespace(name='debian', version='unstable', like_distro='debian'),
+        DistroNamespace(name='ubuntu', version='14.04', like_distro='ubuntu'),
+        DistroNamespace(name='ubuntu', version='14.10', like_distro='ubuntu'),
+        DistroNamespace(name='ubuntu', version='15.03', like_distro='ubuntu'),
+        DistroNamespace(name='ubuntu', version='15.10', like_distro='ubuntu'),
+        DistroNamespace(name='ubuntu', version='16.04', like_distro='ubuntu'),
+        DistroNamespace(name='ubuntu', version='16.10', like_distro='ubuntu'),
+        DistroNamespace(name='ubuntu', version='17.04', like_distro='ubuntu'),
+        DistroNamespace(name='ubuntu', version='17.10', like_distro='ubuntu'),
+        DistroNamespace(name='ubuntu', version='18.04', like_distro='ubuntu'),
+        DistroNamespace(name='ubuntu', version='18.10', like_distro='ubuntu'),
+        DistroNamespace(name='ubuntu', version='19.04', like_distro='ubuntu'),
+        DistroNamespace(name='ubuntu', version='19.10', like_distro='ubuntu')
+    ]
+
+    fail = [
+        DistroNamespace(name='alpine', version='3.1', like_distro='alpine'),
+        DistroNamespace(name='alpine', version='3.1.1', like_distro='alpine'),
+
+        DistroNamespace(name='busybox', version='3', like_distro='busybox'),
+        DistroNamespace(name='linuxmint', version='16', like_distro='debian'),
+        DistroNamespace(name='redhat', version='4', like_distro='rhel'),
+        DistroNamespace(name='redhat', version='5', like_distro='rhel'),
+
+        DistroNamespace(name='ubuntu', version='1.0', like_distro='ubuntu'),
+        DistroNamespace(name='centos', version='1.0', like_distro='ubuntu'),
+        DistroNamespace(name='debian', version='1.0', like_distro='ubuntu'),
+        DistroNamespace(name='rhel', version='1.0', like_distro='ubuntu'),
+        DistroNamespace(name='busybox', version='1.0', like_distro='busybox'),
+        DistroNamespace(name='alpine', version='11.0', like_distro='ubuntu'),
+        DistroNamespace(name='fedora', version='25', like_distro='fedora'),
+        DistroNamespace(name='mageia', version='5', like_distro='mandriva,fedora')
+    ]
+
+    for i in expected:
```

Comment:
doing loops in tests is troublesome because all the variations are treated as a single test. The problem with that is that if item 3 out of 100 fails then it is not possible to know if any of the 97 items are failing as well. Another issue with looping and failures is that it obscures what is passing or testing.

Pytest can help here with parametrize, there will be no need to print out what the failure is and for what namespace, it all gets reported. The API might be a bit awkward if you've not used it before. But roughly, you need to separate the loops and map the items in the lists to each test. This is how it should look:

```python
# Not exhaustive, only for the feeds directly in the test data set
expected = [
    DistroNamespace(name='amzn', version='2', like_distro='amzn'),
    DistroNamespace(name='alpine', version='3.3', like_distro='alpine'),
    DistroNamespace(name='alpine', version='3.4', like_distro='alpine'),
    DistroNamespace(name='alpine', version='3.5', like_distro='alpine'),
    DistroNamespace(name='alpine', version='3.6', like_distro='alpine'),
    DistroNamespace(name='alpine', version='3.7', like_distro='alpine'),
    DistroNamespace(name='alpine', version='3.8', like_distro='alpine'),
    DistroNamespace(name='alpine', version='3.9', like_distro='alpine'),
    DistroNamespace(name='alpine', version='3.10', like_distro='alpine'),
    DistroNamespace(name='alpine', version='3.11', like_distro='alpine'),
    DistroNamespace(name='centos', version='7', like_distro='rhel'),
    DistroNamespace(name='centos', version='7.1', like_distro='rhel'),
    DistroNamespace(name='centos', version='7.3', like_distro='rhel'),
    DistroNamespace(name='centos', version='6', like_distro='rhel'),
    DistroNamespace(name='centos', version='5', like_distro='rhel'),
    DistroNamespace(name='centos', version='8', like_distro='rhel'),
    DistroNamespace(name='centos', version='8.1', like_distro='rhel'),
    DistroNamespace(name='ol', version='7.3', like_distro='ol'),
    DistroNamespace(name='ol', version='6', like_distro='ol'),
    DistroNamespace(name='ol', version='7.3', like_distro='ol'),
    DistroNamespace(name='rhel', version='6', like_distro='rhel'),
    DistroNamespace(name='rhel', version='7', like_distro='rhel'),
    DistroNamespace(name='rhel', version='7.1', like_distro='rhel'),
    DistroNamespace(name='rhel', version='8', like_distro='rhel'),
    DistroNamespace(name='rhel', version='8.1', like_distro='rhel'),
    DistroNamespace(name='debian', version='8', like_distro='debian'),
    DistroNamespace(name='debian', version='9', like_distro='debian'),
    DistroNamespace(name='debian', version='10', like_distro='debian'),
    DistroNamespace(name='debian', version='11', like_distro='debian'),
    DistroNamespace(name='debian', version='unstable', like_distro='debian'),
    DistroNamespace(name='ubuntu', version='14.04', like_distro='ubuntu'),
    DistroNamespace(name='ubuntu', version='14.10', like_distro='ubuntu'),
    DistroNamespace(name='ubuntu', version='15.03', like_distro='ubuntu'),
    DistroNamespace(name='ubuntu', version='15.10', like_distro='ubuntu'),
    DistroNamespace(name='ubuntu', version='16.04', like_distro='ubuntu'),
    DistroNamespace(name='ubuntu', version='16.10', like_distro='ubuntu'),
    DistroNamespace(name='ubuntu', version='17.04', like_distro='ubuntu'),
    DistroNamespace(name='ubuntu', version='17.10', like_distro='ubuntu'),
    DistroNamespace(name='ubuntu', version='18.04', like_distro='ubuntu'),
    DistroNamespace(name='ubuntu', version='18.10', like_distro='ubuntu'),
    DistroNamespace(name='ubuntu', version='19.04', like_distro='ubuntu'),
    DistroNamespace(name='ubuntu', version='19.10', like_distro='ubuntu')
]

fail = [
    DistroNamespace(name='alpine', version='3.1', like_distro='alpine'),
    DistroNamespace(name='alpine', version='3.1.1', like_distro='alpine'),

    DistroNamespace(name='busybox', version='3', like_distro='busybox'),
    DistroNamespace(name='linuxmint', version='16', like_distro='debian'),
    DistroNamespace(name='redhat', version='4', like_distro='rhel'),
    DistroNamespace(name='redhat', version='5', like_distro='rhel'),

    DistroNamespace(name='ubuntu', version='1.0', like_distro='ubuntu'),
    DistroNamespace(name='centos', version='1.0', like_distro='ubuntu'),
    DistroNamespace(name='debian', version='1.0', like_distro='ubuntu'),
    DistroNamespace(name='rhel', version='1.0', like_distro='ubuntu'),
    DistroNamespace(name='busybox', version='1.0', like_distro='busybox'),
    DistroNamespace(name='alpine', version='11.0', like_distro='ubuntu'),
    DistroNamespace(name='fedora', version='25', like_distro='fedora'),
    DistroNamespace(name='mageia', version='5', like_distro='mandriva,fedora')
]

@pytest.mark.parametrize('namespace', fail)
def test_unexpected_namespace_support(initialized_mappings, namespace):
    assert not vulnerabilities.have_vulnerabilities_for(namespace)

@pytest.mark.parametrize('namespace', expected)
def test_expected_namespace_support(initialized_mappings, namespace):
    assert vulnerabilities.have_vulnerabilities_for(namespace)
```

Reply:
ack will make that change. I'm closing this PR and will open a new one with this branch from my fork, which has other updates as well.